example-apps/token_curated_registry/db/mapper.go

```diff
@@ -0,0 +1,161 @@
+package db
+
+import (
+	"github.com/cosmos/cosmos-academy/example-apps/token_curated_registry/types"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/x/bank"
+	"github.com/tendermint/go-amino"
+)
+
+type BallotMapper struct {
+	ListingKey sdk.StoreKey
+
+	CommitKey sdk.StoreKey
+
+	RevealKey sdk.StoreKey
+
+	BallotKey sdk.StoreKey
+
+	Cdc *amino.Codec
+}
+
+func NewBallotMapper(listingKey sdk.StoreKey, ballotkey sdk.StoreKey, commitKey sdk.StoreKey, revealKey sdk.StoreKey, _cdc *amino.Codec) BallotMapper {
```

Comment:
Are you using `mapper` instead of `keeper` because it does not hold any reference to other `keeper`. In any case, it could be useful to include a `keeper` that allows reads to your store for other modules to use.

Reply:
What would a keeper provide over mapper in terms of functionality? My understanding was that the keeper was just a convenience wrapper over the mapper/

Reply:
Citing jae: "Mappers and Keepers provide access to KV stores via the context. The only
difference between the two is that a Mapper provides a lower-level API, so
generally a Keeper might hold references to other Keepers and Mappers but not
vice versa."